src/m_menu.h

```diff
@@ -184,6 +184,9 @@ typedef struct setup_menu_s
   const char **selectstrings; // [FG] selection of choices
 } setup_menu_t;
 
+extern int menu_background;
+extern boolean setup_active;
```

Comment:
Exposing internal variables like `setup_active` can be a bit dangerous. It couples the code outside to the specific details of the menu system, which means any changes in one place will require changes everywhere else. It also makes the code more susceptible to bugs introduced when one part of the code ends up using the variable in an unexpected way. I'd recommend setting up a method to query the information you want with clear semantics and exposing the method instead. This is an example of that kind of thing in dsda-doom:
```c
dboolean M_ConsoleOpen(void)
{
  return menuactive && currentMenu == &dsda_ConsoleDef;
}
```

Reply:
Something like this?
```c
boolean M_IsSetupActive(void)
{
  return setup_active; // Should we also check for menuactive?
}
```

Reply:
I guess what Ryan means is something like
```
void M_MenuIsShaded(void)
{
  return setup_active && menu_background == 2;
}
```

Reply:
The goal with something like this is to isolate as much information as possible and focus on the question you're trying to answer. This is why Fabian's function wraps the whole thing. The code outside doesn't need to know about how the menu tracks what is open or what value of the menu background corresponds to shading (or how it's tracked). It just wants to know if it's shaded. This also has the benefit of making the code easier to digest.